template/netcore30/Program.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using Function;
+
+namespace root
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string buffer = Console.In.ReadToEnd();

```

Comment:
Have you considered changing to to ```static async Task Main(string[] args)``` and changing to ```await Console.In.ReadToEndAsync();```?  If you do this you could also change the FunctionHandler to be async.

Reply:
I copy the code, but I think it some potentials to optimize the implementations before the migrations. Should we consider to append different templates like we talking on slack? -http, -kestrel, -async?